### DISCUSSION TITLE
Override the PHP file of database notifications drawer livewire component

### DISCUSSION THREAD

**Discussion by dev-dev-15** [filamentphp/filament#8932] [Help]
I want to override the PHP file of this livewire component (database notifications drawer):
![image](https://github.com/filamentphp/filament/assets/14108109/ca1a171a-1c0f-4485-9e2c-3982a3bb2d3d)

What I want is to edit the actions of the buttons on the drawer (Clear, Mark all as read...)
The class in in vendor/Filament/Notifications/src/Livewire, it's called DatabaseNotifications.php

**Comment by krekas** [comment]
What exactly you want to change?

**Reply by dev-dev-15** [reply]
I want to edit the actions of the buttons: Mark all as read, clear and the X button of the notification, basically what I want is to overwrite the class vendor/Filament/Notifications/src/Livewire/DatabaseNotifications.php, which is the PHP file of the notification drawer livewire component.
Reason: those buttons are are using mass assignment to change the notifications which means no events will be triggered but I want the events to be triggered.
